application/src/main/java/org/togetherjava/tjbot/commands/tag/TagManageCommand.java

```diff
@@ -0,0 +1,233 @@
+                new SubcommandData("createidtag", "Create a tag using a message id")
+                    .addOption(OptionType.STRING, TAG_MINUS_ID, TAG_ID, true)
+                    .addOption(OptionType.NUMBER, MESSAGE_MINUS_ID, "Message id", true));
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandEvent event) {
+        Member member = event.getMember();
+
+        if (!member.hasPermission(Permission.MESSAGE_MANAGE)) {
+            event.reply("You need the MESSAGE_MANAGE permission to use this command!")
+                .setEphemeral(true)
+                .queue();
+
+            return;
+        }
+
+        switch (event.getSubcommandName()) {
+            case "edittag" -> editTag(event);
+            case "editidtag" -> editIdTag(event);
+            case "deletetag" -> deleteTag(event, member);
+            case "createtag" -> createTag(event);
+            case "createidtag" -> createIdTag(event);
+            default -> throw new IllegalStateException();
+        }
+    }
+
+    private void editTag(SlashCommandEvent event) {
+        String tagId = event.getOption("id").getAsString();
+        String content = event.getOption(CONTENT).getAsString();
+
+        if (!tagSystem.exists(tagId)) {
+            event.reply(THIS_TAG_DOES_NOT_EXIST).setEphemeral(true).queue();
+
+            return;
+        }
+
+        tagSystem.put(tagId, content);
+
+        event
+            .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                .setTitle(SUCCESS)
+                .setTimestamp(LocalDateTime.now())
+                .setFooter(event.getUser().getAsTag())
+                .setDescription("Successfully edited tag '" + tagId + "'")
+                .build())
+            .queue();
+    }
+
+    private void editIdTag(SlashCommandEvent event) {
+        String tagId = event.getOption(TAG_MINUS_ID).getAsString();
+        long messageId = event.getOption(MESSAGE_MINUS_ID).getAsLong();
+
+        if (!tagSystem.exists(tagId)) {
+            event.reply(THIS_TAG_DOES_NOT_EXIST).setEphemeral(true).queue();
+
+            return;
+        }
+
+        event.getMessageChannel().retrieveMessageById(messageId).queue(message -> {
+            tagSystem.put(tagId, message.getContentRaw());
+
+            event
+                .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                    .setTitle(SUCCESS)
+                    .setTimestamp(LocalDateTime.now())
+                    .setFooter(event.getUser().getAsTag())
+                    .setDescription("Successfully edited tag '" + tagId + "'")
+                    .build())
+                .queue();
+        }, failure -> {
+            if (failure instanceof ErrorResponseException ex
+                    && ex.getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
+                event.reply("This message doesn't exist").setEphemeral(true).queue();
+            } else {
+                throw new RuntimeException(failure);
```

Comment:
You can also just print the existing failure, throwing a new runtime exception is kinda useless?

Reply:
agreed. not only that, `RuntimeException` also does not add anything useful to the table.

either use a more specific exception, add error details to it or just rethrow instead. `throw failure;`.

Reply:
I can't directly rethrow it here, since the 2nd argument is a consumer.

Reply:
I don't understand why you're throwing the same exception within a RuntimeException?
Instead, just print the error and tell the user something went wrong?

Reply:
Yea, I'm just printing the stacktrace of the exception now.

Reply:
Fair, saw this discussion just now (disregard my new comment on this topic please).

Still, use the logging system instead of `printStackTrace` instead, if applicable.